### ISSUE TITLE
Default route gets resolved in test code when calling $httpBackend.flush() or scope.$apply()

### ISSUE THREAD

**Issue by achydenius** [angular/angular.js#14337]
**Do you want to request a *feature* or report a *bug*?**

Report a breaking change (or a bug).

**What is the current behavior?**

When upgrading from 1.5.0 to 1.5.1 or newer, calling `$httpBackend.flush()` or `scope.$apply()` in test code causes the default route to resolve automatically. This can lead unexpected requests and break the tests. This seems to be due to change in [ngRoute](https://github.com/angular/angular.js/commit/8237482d49e76e2c4994fe6207e3c9799ef04163).

**If the current behavior is a bug, please provide the steps to reproduce**

Can be reproduced with for example [AngularJS Phonecat demo](https://github.com/angular/angular-phonecat) by updating bower.json to use Angular 1.5.1 and running the tests.

**Comment by gkalpak** [comment]
Hm...this can be indeed inconvenient :confused:
Basically, this is caused by either specifying a `/` route (`$routeProvider.when('/', ...)`) or redirecting to some "fallback" route (`$routeProvider.otherwise(...)`).

It is possible to work around this (in several ugly ways). The two most straight-forward ways out are to either add a dummy template to the `$templateCache` or add an expectation for the template URL:

```js
beforeEach(module(function () {
  return function ($templateCache) {
    $templateCache.put('/default/route/template/url', '');
  };
}));

// or

beforeEach(module(function () {
  return function ($httpBackend) {
    $httpBackend.whenGET('/default/route/template/url').respond('');
  };
}));
```

--
FWIW, ui-router does a very similar fix as 8237482.
Maybe we should revert it on `v1.5.x` and leave it on `v1.6` as a breaking change.

[1]: https://codepen.io/ExpertSystem/pen/pyrjbz?editors=0010

**Comment by gkalpak** [comment]
Of course, a more proper solution is not to include `ngRoute` (or a module that depends on it) in tests that don't need it :smiley:
